tests/driver_lis2dh12/main.c

```diff
@@ -11,170 +11,682 @@
+        gpio_irq_enable(dev.p->int1_pin);
+        gpio_irq_enable(dev.p->int2_pin);
+
+        /* thread will sleep until next wokeup_lis */
+        thread_sleep();
+    }
+
+    return NULL;
+}
+#endif /* MODULE_LIS2DH12_INT */
+
+int shell_lis2dh12_cmd(int argc, char **argv) {
+
+    printf("Command: lis %s %s\n", (argc > 1) ? argv[1] : "",
+           (argc > 2) ? argv[2] : "");
+#ifdef MODULE_LIS2DH12_INT
+    const char * usage = "USAGE: lis <subcommand> [arg], with subcommand "
+                         "in (enable, disable, read, read_fifo, clear_data, "
+                         "set-click, set-thold-shock, set-thold-inter, "
+                         "set-highpass, change_rate, change_power, change_scale).";
 #else
-        xtimer_usleep(DELAY);
-#endif
+    const char * usage = "USAGE: lis <subcommand> [arg], with subcommand "
+                         "in (enable, disable, read, read_fifo, clear_data, "
+                         "change_rate, change_power, change_scale).";
+#endif /* MODULE_LIS2DH12_INT */
+
+    /* MISSING command */
+    if (argc < 2) {
+        printf("Error: Missing sub-command. %s\n", usage);
+        return -1;
+    }
+
+    /* enable disable device */
+    else if (strncmp(argv[1], "enable", sizeof("enable")) == 0) {
+        if (lis2dh12_poweron(&dev) != LIS2DH12_OK) {
+            puts("unable to poweron device.");
+        }
+        return 1;
+    }
+    else if (strncmp(argv[1], "disable", sizeof("disable")) == 0) {
+        if (lis2dh12_poweroff(&dev) != LIS2DH12_OK) {
+            puts("unable to poweroff device.");
+            return -1;
+        }
+        return 1;
+    }
+
+    /* read acceleration data */
+    else if (strncmp(argv[1], "read", sizeof("read")) == 0) {
+        uint8_t amount = (argc < 3) ? 1 : atoi(argv[2]);
+        uint8_t amt = 0;
 
         /* read sensor data */
-        int16_t data[3];
-        if (lis2dh12_read(&dev, data) != LIS2DH12_OK) {
-            puts("error: unable to retrieve data from sensor");
-            continue;
+        for (amt = 0; amt < amount; amt++){
+            if (lis2dh12_read(&dev, data_lis) != LIS2DH12_OK) {
+                puts("error: no data from sensor");
+                return -1;
+            }
+            /* format data */
+            for (int i = 0; i < 3; i++) {
+                size_t len = fmt_s16_dfp(str_out[i], data_lis[i], -3);
+                str_out[i][len] = '\0';
+            }
+
+            /* print data to STDIO */
+            printf("X: %6s  Y: %6s  Z: %6s\n", str_out[0], str_out[1], str_out[2]);
+
+            xtimer_msleep(250);
+        }
+        return 1;
+    }
+    else if (strncmp(argv[1], "read_fifo", sizeof("read_fifo")) == 0) {
+        uint8_t number = 0;
+        if ((argc < 3) || (number = atoi(argv[2])) > 32) {
+            puts("Error: Missing parameter. The command should "
+                 "contain number of FIFO values to read (max. 32)). "
+                 "USAGE: lis read_fifo [number]");
```

Comment:
If you have to break up the lines to look pretty in the code, you should also break them up in the output.

```suggestion
            puts("Error: Missing parameter.")
            puts("The command should contain number of FIFO values to read (max. 32)).");
            puts("USAGE: lis read_fifo [number]");
```